nixos/modules/services/networking/firewall.nix

```diff
@@ -493,6 +508,22 @@ in
         '';
     };
 
+    networking.firewall.extraAllowed = mkOption {
+      type = types.listOf (types.submodule { options = {
+           source = mkOption { type = types.string; default = ""; example = "example.com"; description = "Source of the packets"; };
+           protocol = mkOption { type = types.enum ["tcp" "udp" "icmp"]; default = ""; example = "tcp"; description = "Protocol used."; };
+           dport = mkOption { type = types.int; default = 0; example = 22; description = "Destination port of the packet."; };
+           useIPv4 = mkOption { type = types.bool; default = true; description = "Use the IPv4 address of this host."; };
+           useIPv6 = mkOption { type = types.bool; default = true; description = "Use the IPv6 address of this host."; };
+      };});
+      default = [];
+      example = [ { source = "example.com"; protocol = "tcp"; dport = 22; } ];
```

Comment:
From the `iptables` manpage:
```
Hostnames will be resolved once only, before the rule is submitted to the kernel.  Please note that
specifying any name to be resolved with a remote query such as DNS is a really bad idea.
```
So in order not to give people bad ideas, it would be better to use an IP address in the example. (See https://tools.ietf.org/search/rfc5737)